In [1]:
#Task 1
#import pandas
import pandas as pd
import matplotlib.pyplot as plt
from matplotlib import rcParams
import seaborn as sns
import numpy as np
import os
import sklearn


##import inforamtion from Kaggle 
CCLoan = pd.read_csv(r'D:\springboard\DataScienceGuidedCapstone\Unit7\DataCapstoneProject2\ccFraudDefault\application_data.csv',header=0)

In [2]:
#Task 2
#preview first 25 row to verify data was imported correctly

CCLoan.head(25)

SK_ID_CURR  TARGET NAME_CONTRACT_TYPE CODE_GENDER FLAG_OWN_CAR  \
0       100002       1         Cash loans           M            N   
1       100003       0         Cash loans           F            N   
2       100004       0    Revolving loans           M            Y   
3       100006       0         Cash loans           F            N   
4       100007       0         Cash loans           M            N   
5       100008       0         Cash loans           M            N   
6       100009       0         Cash loans           F            Y   
7       100010       0         Cash loans           M            Y   
8       100011       0         Cash loans           F            N   
9       100012       0    Revolving loans           M            N   
10      100014       0         Cash loans           F            N   
11      100015       0         Cash loans           F            N   
12      100016       0         Cash loans           F            N   
13      100017       0         Cash loans           M            Y   
14      100018       0         Cash loans           F            N   
15      100019       0         Cash loans           M            Y   
16      100020       0         Cash loans           M            N   
17      100021       0    Revolving loans           F            N   
18      100022       0    Revolving loans           F            N   
19      100023       0         Cash loans           F            N   
20      100024       0    Revolving loans           M            Y   
21      100025       0         Cash loans           F            Y   
22      100026       0         Cash loans           F            N   
23      100027       0         Cash loans           F            N   
24      100029       0         Cash loans           M            Y   

   FLAG_OWN_REALTY  CNT_CHILDREN  AMT_INCOME_TOTAL  AMT_CREDIT  AMT_ANNUITY  \
0                Y             0        202500.000    406597.5      24700.5   
1                N             0        270000.000   1293502.5      35698.5   
2                Y             0         67500.000    135000.0       6750.0   
3                Y             0        135000.000    312682.5      29686.5   
4                Y             0        121500.000    513000.0      21865.5   
5                Y             0         99000.000    490495.5      27517.5   
6                Y             1        171000.000   1560726.0      41301.0   
7                Y             0        360000.000   1530000.0      42075.0   
8                Y             0        112500.000   1019610.0      33826.5   
9                Y             0        135000.000    405000.0      20250.0   
10               Y             1        112500.000    652500.0      21177.0   
11               Y             0         38419.155    148365.0      10678.5   
12               Y             0         67500.000     80865.0       5881.5   
13               N             1        225000.000    918468.0      28966.5   
14               Y             0        189000.000    773680.5      32778.0   
15               Y             0        157500.000    299772.0      20160.0   
16               N             0        108000.000    509602.5      26149.5   
17               Y             1         81000.000    270000.0      13500.0   
18               Y             0        112500.000    157500.0       7875.0   
19               Y             1         90000.000    544491.0      17563.5   
20               Y             0        135000.000    427500.0      21375.0   
21               Y             1        202500.000   1132573.5      37561.5   
22               N             1        450000.000    497520.0      32521.5   
23               Y             0         83250.000    239850.0      23850.0   
24               N             2        135000.000    247500.0      12703.5   

    ...  FLAG_DOCUMENT_18 FLAG_DOCUMENT_19 FLAG_DOCUMENT_20 FLAG_DOCUMENT_21  \
0   ...                 0                0      

In [3]:
#Task 3
#Total row of record is 307511
CCLoan.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 307511 entries, 0 to 307510
Columns: 122 entries, SK_ID_CURR to AMT_REQ_CREDIT_BUREAU_YEAR
dtypes: float64(65), int64(41), object(16)
memory usage: 286.2+ MB


In [4]:
#Task 4
#rename certian column to be more readable
CCLoan.rename(columns={'SK_ID_CURR':'CLIENTID','TARGET':'LATE_PAYMENT'},inplace = True)


In [5]:
#Task 5
CCLoan.head(25)


CLIENTID  LATE_PAYMENT NAME_CONTRACT_TYPE CODE_GENDER FLAG_OWN_CAR  \
0     100002             1         Cash loans           M            N   
1     100003             0         Cash loans           F            N   
2     100004             0    Revolving loans           M            Y   
3     100006             0         Cash loans           F            N   
4     100007             0         Cash loans           M            N   
5     100008             0         Cash loans           M            N   
6     100009             0         Cash loans           F            Y   
7     100010             0         Cash loans           M            Y   
8     100011             0         Cash loans           F            N   
9     100012             0    Revolving loans           M            N   
10    100014             0         Cash loans           F            N   
11    100015             0         Cash loans           F            N   
12    100016             0         Cash loans           F            N   
13    100017             0         Cash loans           M            Y   
14    100018             0         Cash loans           F            N   
15    100019             0         Cash loans           M            Y   
16    100020             0         Cash loans           M            N   
17    100021             0    Revolving loans           F            N   
18    100022             0    Revolving loans           F            N   
19    100023             0         Cash loans           F            N   
20    100024             0    Revolving loans           M            Y   
21    100025             0         Cash loans           F            Y   
22    100026             0         Cash loans           F            N   
23    100027             0         Cash loans           F            N   
24    100029             0         Cash loans           M            Y   

   FLAG_OWN_REALTY  CNT_CHILDREN  AMT_INCOME_TOTAL  AMT_CREDIT  AMT_ANNUITY  \
0                Y             0        202500.000    406597.5      24700.5   
1                N             0        270000.000   1293502.5      35698.5   
2                Y             0         67500.000    135000.0       6750.0   
3                Y             0        135000.000    312682.5      29686.5   
4                Y             0        121500.000    513000.0      21865.5   
5                Y             0         99000.000    490495.5      27517.5   
6                Y             1        171000.000   1560726.0      41301.0   
7                Y             0        360000.000   1530000.0      42075.0   
8                Y             0        112500.000   1019610.0      33826.5   
9                Y             0        135000.000    405000.0      20250.0   
10               Y             1        112500.000    652500.0      21177.0   
11               Y             0         38419.155    148365.0      10678.5   
12               Y             0         67500.000     80865.0       5881.5   
13               N             1        225000.000    918468.0      28966.5   
14               Y             0        189000.000    773680.5      32778.0   
15               Y             0        157500.000    299772.0      20160.0   
16               N             0        108000.000    509602.5      26149.5   
17               Y             1         81000.000    270000.0      13500.0   
18               Y             0        112500.000    157500.0       7875.0   
19               Y             1         90000.000    544491.0      17563.5   
20               Y             0        135000.000    427500.0      21375.0   
21               Y             1        202500.000   1132573.5      37561.5   
22               N             1        450000.000    497520.0      32521.5   
23               Y             0         83250.000    239850.0      23850.0   
24               N             2        135000.000    247500.0      12703.5   

    ...  FLAG_DOCUMENT_1

In [6]:
#Task 5.1
CCLoan.describe()


CLIENTID   LATE_PAYMENT   CNT_CHILDREN  AMT_INCOME_TOTAL  \
count  307511.000000  307511.000000  307511.000000      3.075110e+05   
mean   278180.518577       0.080729       0.417052      1.687979e+05   
std    102790.175348       0.272419       0.722121      2.371231e+05   
min    100002.000000       0.000000       0.000000      2.565000e+04   
25%    189145.500000       0.000000       0.000000      1.125000e+05   
50%    278202.000000       0.000000       0.000000      1.471500e+05   
75%    367142.500000       0.000000       1.000000      2.025000e+05   
max    456255.000000       1.000000      19.000000      1.170000e+08   

         AMT_CREDIT    AMT_ANNUITY  AMT_GOODS_PRICE  \
count  3.075110e+05  307499.000000     3.072330e+05   
mean   5.990260e+05   27108.573909     5.383962e+05   
std    4.024908e+05   14493.737315     3.694465e+05   
min    4.500000e+04    1615.500000     4.050000e+04   
25%    2.700000e+05   16524.000000     2.385000e+05   
50%    5.135310e+05   24903.000000     4.500000e+05   
75%    8.086500e+05   34596.000000     6.795000e+05   
max    4.050000e+06  258025.500000     4.050000e+06   

       REGION_POPULATION_RELATIVE     DAYS_BIRTH  DAYS_EMPLOYED  ...  \
count               307511.000000  307511.000000  307511.000000  ...   
mean                     0.020868  -16036.995067   63815.045904  ...   
std                      0.013831    4363.988632  141275.766519  ...   
min                      0.000290  -25229.000000  -17912.000000  ...   
25%                      0.010006  -19682.000000   -2760.000000  ...   
50%                      0.018850  -15750.000000   -1213.000000  ...   
75%                      0.028663  -12413.000000    -289.000000  ...   
max                      0.072508   -7489.000000  365243.000000  ...   

       FLAG_DOCUMENT_18  FLAG_DOCUMENT_19  FLAG_DOCUMENT_20  FLAG_DOCUMENT_21  \
count     307511.000000     307511.000000     307511.000000     307511.000000   
mean           0.008130          0.000595          0.000507          0.000335   
std            0.089798          0.024387          0.022518          0.018299   
min            0.000000          0.000000          0.000000          0.000000   
25%            0.000000          0.000000          0.000000          0.000000   
50%            0.000000          0.000000          0.000000          0.000000   
75%            0.000000          0.000000          0.000000          0.000000   
max            1.000000          1.000000          1.000000          1.000000   

       AMT_REQ_CREDIT_BUREAU_HOUR  AMT_REQ_CREDIT_BUREAU_DAY  \
count               265992.000000              265992.000000   
mean                     0.006402                   0.007000   
std                      0.083849                   0.110757   
min                      0.000000                   0.000000   
25%                      0.000000                   0.000000   
50%                      0.000000                   0.000000   
75%                      0.000000                   0.000000   
max                      4.000000                   9.000000   

       AMT_REQ_CREDIT_BUREAU_WEEK  AMT_REQ_CREDIT_BUREAU_MON  \
count               265992.000000              265992.000000   
mean                     0.034362                   0.267395   
std                      0.204685                   0.916002   
min                      0.000000                   0.000000   
25%                      0.000000                   0.000000   
50%                      0.000000                   0.000000   
75%                      0.000000                   0.000000   
max                      8.000000                  27.000000   

       AMT_REQ_CREDIT_BUREAU_QRT  AMT_REQ_CREDIT_BUREAU_YEAR  
count              265992.000000               265992.000000  
mean                    0.265474                    1.899974  
std                     0.794056                    1.869295  
min                     0.000000                    0.000000  
25%                

In this section 5.2.x i will look at each of the 122 column and see the number null value.  It will be better if most columns are not null and have NaN

In [7]:
#Task 5.2.1
#Checking first 1-13 column of missing data 

CC_Late = CCLoan['LATE_PAYMENT'].isnull()
print ('Late Payment Missing: '+str(CC_Late.sum()))

CC_CONTRACT = CCLoan['NAME_CONTRACT_TYPE'].isnull()
print ('Contract Type Missing: '+str(CC_CONTRACT.sum()))

CC_GENDER=CCLoan['CODE_GENDER'].isnull()
print ('Gender Missing: '+str(CC_GENDER.sum()))

CC_OWNCAR=CCLoan['FLAG_OWN_CAR'].isnull()
print ('Own Car Missing: '+str(CC_OWNCAR.sum()))

CC_OWNREALITY=CCLoan['FLAG_OWN_REALTY'].isnull()
print ('Own Reality Missing: '+str(CC_OWNREALITY.sum()))

CC_CHILD =CCLoan['CNT_CHILDREN'].isnull()
print ('Own Reality Missing: '+str(CC_OWNREALITY.sum()))

CC_INCOME=CCLoan['AMT_INCOME_TOTAL'].isnull()
print ('Own Reality Missing: '+str(CC_INCOME.sum()))

CC_CREDIT=CCLoan['AMT_CREDIT'].isnull()
print ('Own Reality Missing: '+str(CC_CREDIT.sum()))

CC_ANNUITY=CCLoan['AMT_ANNUITY'].isnull()
print ('Annuity Missing: '+str(CC_ANNUITY.sum()))

CC_GOODSPRICE=CCLoan['AMT_GOODS_PRICE'].isnull()
print ('Goods Price Missing: '+str(CC_GOODSPRICE.sum()))




Late Payment Missing: 0
Contract Type Missing: 0
Gender Missing: 0
Own Car Missing: 0
Own Reality Missing: 0
Own Reality Missing: 0
Own Reality Missing: 0
Own Reality Missing: 0
Annuity Missing: 12
Goods Price Missing: 278


In [8]:
#Task 5.2.1.1
#show the 12 AMT_ANNUITY is missing. 
#Looking at data says it is NaN.  So it is fine
CC_ANNUITY1 = CCLoan.loc[CCLoan['AMT_ANNUITY'].isnull()]
CC_ANNUITY1 [['CLIENTID','AMT_INCOME_TOTAL','AMT_CREDIT','AMT_ANNUITY','AMT_GOODS_PRICE']].head(25)

CLIENTID  AMT_INCOME_TOTAL  AMT_CREDIT  AMT_ANNUITY  AMT_GOODS_PRICE
47531     155054          180000.0    450000.0          NaN         450000.0
50035     157917           94500.0    450000.0          NaN         450000.0
51594     159744          202500.0    539100.0          NaN         450000.0
55025     163757          162000.0    296280.0          NaN         225000.0
59934     169487          202500.0    360000.0          NaN         360000.0
75873     187985          144000.0    219249.0          NaN         166500.0
89343     203726           90000.0    157500.0          NaN         157500.0
123872    243648          202500.0    929088.0          NaN         720000.0
207186    340147          171000.0    486000.0          NaN         486000.0
227939    364022          315000.0    628069.5          NaN         499500.0
239329    377174          157500.0    792000.0          NaN         792000.0
241835    379997          315000.0   1483231.5          NaN        1354500.0

In [9]:
#Task 5.2.1.2
#It is not missing only NaN
CC_GOODSPRICE1= CCLoan.loc[CCLoan['AMT_GOODS_PRICE'].isnull()]
CC_GOODSPRICE1 [['CLIENTID','AMT_INCOME_TOTAL','AMT_CREDIT','AMT_ANNUITY','AMT_GOODS_PRICE']].head(25)
#check if there is zero price(mean price) and price likely 0

CLIENTID  AMT_INCOME_TOTAL  AMT_CREDIT  AMT_ANNUITY  AMT_GOODS_PRICE
724      100837           45000.0    135000.0       6750.0              NaN
5937     106955          157500.0    450000.0      22500.0              NaN
6425     107494           67500.0    202500.0      10125.0              NaN
6703     107822          121500.0    180000.0       9000.0              NaN
7647     108913          180000.0    450000.0      22500.0              NaN
7880     109190          121500.0    270000.0      13500.0              NaN
7995     109322          112500.0    180000.0       9000.0              NaN
10819    112595           90000.0    270000.0      13500.0              NaN
11287    113148          225000.0    135000.0       6750.0              NaN
13008    115162          157500.0    450000.0      22500.0              NaN
14699    117150          135000.0    270000.0      13500.0              NaN
15953    118618           90000.0    270000.0      13500.0              NaN
18935    122080          157500.0    180000.0       9000.0              NaN
19178    122374           67500.0    180000.0       9000.0              NaN
19921    123233           90000.0    180000.0       9000.0              NaN
21193    124697          202500.0    270000.0      13500.0              NaN
21338    124859          225000.0    675000.0      33750.0              NaN
21546    125096           45000.0    135000.0       6750.0              NaN
25391    129532          382500.0    270000.0      13500.0              NaN
26398    130692          135000.0    495000.0      24750.0              NaN
26736    131077           90000.0    202500.0      10125.0              NaN
27003    131385          180000.0    450000.0      22500.0              NaN
28201    132778           45000.0    315000.0      15750.0              NaN
29059    133761           81000.0    247500.0      12375.0              NaN
30164    135010          121500.0    405000.0      20250.0              NaN

In [10]:
#Task 5.2.2
#Checking next 14-23 column of missing data 

CC_TYPESUIT = CCLoan['NAME_TYPE_SUITE'].isnull()
print ('NAME_TYPE_SUITE: '+str(CC_TYPESUIT.sum()))

CC_INCOMETYPE = CCLoan['NAME_INCOME_TYPE'].isnull()
print ('NAME_INCOME_TYPE Missing: '+str(CC_INCOMETYPE.sum()))

CC_EDUCATIONTYPE=CCLoan['NAME_EDUCATION_TYPE'].isnull()
print ('NAME_EDUCATION_TYPE Missing: '+str(CC_EDUCATIONTYPE.sum()))

CC_FAMILY=CCLoan['NAME_FAMILY_STATUS'].isnull()
print ('NAME_FAMILY_STATUS Missing: '+str(CC_FAMILY.sum()))

CC_HOUSETYPE=CCLoan['NAME_HOUSING_TYPE'].isnull()
print ('NAME_HOUSING_TYPE Missing: '+str(CC_HOUSETYPE.sum()))

CC_REGION_POPULATION =CCLoan['REGION_POPULATION_RELATIVE'].isnull()
print ('REGION_POPULATION_RELATIVE Missing: '+str(CC_REGION_POPULATION.sum()))

CC_DAYSBIRTH=CCLoan['DAYS_BIRTH'].isnull()
print ('DAYS_BIRTH Missing: '+str(CC_DAYSBIRTH.sum()))

CC_EMPLOYED=CCLoan['DAYS_EMPLOYED'].isnull()
print ('DAYS_EMPLOYED Missing: '+str(CC_EMPLOYED.sum()))

CC_ANNUITY=CCLoan['DAYS_REGISTRATION'].isnull()
print ('ANNUITY Missing: '+str(CC_ANNUITY.sum()))

CC_GOODSPRICE=CCLoan['DAYS_ID_PUBLISH'].isnull()
print ('Goods Price Missing: '+str(CC_GOODSPRICE.sum()))

NAME_TYPE_SUITE: 1292
NAME_INCOME_TYPE Missing: 0
NAME_EDUCATION_TYPE Missing: 0
NAME_FAMILY_STATUS Missing: 0
NAME_HOUSING_TYPE Missing: 0
REGION_POPULATION_RELATIVE Missing: 0
DAYS_BIRTH Missing: 0
DAYS_EMPLOYED Missing: 0
ANNUITY Missing: 0
Goods Price Missing: 0


In [11]:
#Task 5.2.2.1
#It is not missing only NaN because maybe no one came with them to borrow loan
CC_NAMETYPE1= CCLoan.loc[CCLoan['NAME_TYPE_SUITE'].isnull()]
CC_NAMETYPE1 [['CLIENTID','AMT_INCOME_TOTAL','AMT_CREDIT','AMT_ANNUITY','NAME_TYPE_SUITE']].head(25)
#rename 

CLIENTID  AMT_INCOME_TOTAL  AMT_CREDIT  AMT_ANNUITY NAME_TYPE_SUITE
64      100076          180000.0    315000.0       9679.5             NaN
724     100837           45000.0    135000.0       6750.0             NaN
1199    101414          135000.0    533668.5      34636.5             NaN
1669    101954          153000.0   1162170.0      30654.0             NaN
1862    102183          121500.0    268659.0      13774.5             NaN
2271    102661          225000.0   1890000.0      49986.0             NaN
2625    103065          112500.0    593010.0      16434.0             NaN
2650    103093          360000.0    728460.0      37134.0             NaN
3012    103519          135000.0    675000.0      17806.5             NaN
3050    103560          225000.0   1506816.0      45814.5             NaN
3371    103938         1350000.0   2410380.0     109053.0             NaN
3718    104339          225000.0    675000.0      20596.5             NaN
4063    104754          112500.0    373311.0      13963.5             NaN
4193    104904          157500.0    585000.0      16218.0             NaN
4240    104962           90000.0    327024.0      10575.0             NaN
4338    105076          315000.0   1762110.0      48586.5             NaN
4428    105177          195750.0    225000.0      14764.5             NaN
4669    105460          135000.0    755190.0      28116.0             NaN
4749    105551          166500.0    239850.0      22540.5             NaN
4876    105701          225000.0   1078200.0      29781.0             NaN
5304    106200          117000.0    781920.0      23836.5             NaN
5786    106766          270000.0   1350000.0      35743.5             NaN
5804    106785          157500.0    808650.0      23305.5             NaN
5937    106955          157500.0    450000.0      22500.0             NaN
6425    107494           67500.0    202500.0      10125.0             NaN

In [12]:
#Task 5.2.3
#Checking next 24-33 column of missing data 

CC_CARAGE = CCLoan['OWN_CAR_AGE'].isnull()
print ('OWN_CAR_AGE Missing: '+str(CC_CARAGE.sum()))

CC_MOBIL = CCLoan['FLAG_MOBIL'].isnull()
print ('CC_MOBIL Missing: '+str(CC_MOBIL.sum()))

CC_EMPPHONE=CCLoan['FLAG_EMP_PHONE'].isnull()
print ('CC_EMPPHONE Missing: '+str(CC_EMPPHONE.sum()))

CC_WORKPHONE=CCLoan['FLAG_WORK_PHONE'].isnull()
print ('CC_WORKPHONE: '+str(CC_WORKPHONE.sum()))

CC_CONTMOBILE=CCLoan['FLAG_CONT_MOBILE'].isnull()
print ('CC_CONTMOBILE Missing: '+str(CC_CONTMOBILE.sum()))

CC_PHONE =CCLoan['FLAG_PHONE'].isnull()
print ('CC_PHONE Missing: '+str(CC_PHONE.sum()))

CC_EMAIL=CCLoan['FLAG_EMAIL'].isnull()
print ('CC_EMAIL Missing: '+str(CC_EMAIL.sum()))

CC_OCCUPATIONTYPE=CCLoan['OCCUPATION_TYPE'].isnull()
print ('CC_OCCUPATIONTYPE Missing: '+str(CC_OCCUPATIONTYPE.sum()))

CC_FAMMEMEBER=CCLoan['CNT_FAM_MEMBERS'].isnull()
print ('CC_FAMMEMEBER Missing: '+str(CC_FAMMEMEBER.sum()))

CC_CLIENTRATING=CCLoan['REGION_RATING_CLIENT'].isnull()
print ('CC_CLIENTRATING Missing: '+str(CC_CLIENTRATING.sum()))

OWN_CAR_AGE Missing: 202929
CC_MOBIL Missing: 0
CC_EMPPHONE Missing: 0
CC_WORKPHONE: 0
CC_CONTMOBILE Missing: 0
CC_PHONE Missing: 0
CC_EMAIL Missing: 0
CC_OCCUPATIONTYPE Missing: 96391
CC_FAMMEMEBER Missing: 2
CC_CLIENTRATING Missing: 0


In [13]:
#Task 5.2.3.1
#It is not missing only NaN because maybe no one came with them to borrow loan
CC_OWN_CAR_AGE1= CCLoan.loc[CCLoan['OWN_CAR_AGE'].isnull()]
CC_OWN_CAR_AGE1 [['CLIENTID','AMT_INCOME_TOTAL','AMT_CREDIT','AMT_ANNUITY','OWN_CAR_AGE']].head(25)

CLIENTID  AMT_INCOME_TOTAL  AMT_CREDIT  AMT_ANNUITY  OWN_CAR_AGE
0     100002        202500.000    406597.5      24700.5          NaN
1     100003        270000.000   1293502.5      35698.5          NaN
3     100006        135000.000    312682.5      29686.5          NaN
4     100007        121500.000    513000.0      21865.5          NaN
5     100008         99000.000    490495.5      27517.5          NaN
8     100011        112500.000   1019610.0      33826.5          NaN
9     100012        135000.000    405000.0      20250.0          NaN
10    100014        112500.000    652500.0      21177.0          NaN
11    100015         38419.155    148365.0      10678.5          NaN
12    100016         67500.000     80865.0       5881.5          NaN
14    100018        189000.000    773680.5      32778.0          NaN
16    100020        108000.000    509602.5      26149.5          NaN
17    100021         81000.000    270000.0      13500.0          NaN
18    100022        112500.000    157500.0       7875.0          NaN
19    100023         90000.000    544491.0      17563.5          NaN
22    100026        450000.000    497520.0      32521.5          NaN
23    100027         83250.000    239850.0      23850.0          NaN
25    100030         90000.000    225000.0      11074.5          NaN
26    100031        112500.000    979992.0      27076.5          NaN
27    100032        112500.000    327024.0      23827.5          NaN
29    100034         90000.000    180000.0       9000.0          NaN
30    100035        292500.000    665892.0      24592.5          NaN
31    100036        112500.000    512064.0      25033.5          NaN
32    100037         90000.000    199008.0      20893.5          NaN
34    100040        135000.000   1125000.0      32895.0          NaN

In [14]:
#Task 5.2.3.2
#It is not missing only NaN because maybe no one came with them to borrow loan
CC_OCCUPATION_TYPE1= CCLoan.loc[CCLoan['OCCUPATION_TYPE'].isnull()]
CC_OCCUPATION_TYPE1 [['CLIENTID','AMT_INCOME_TOTAL','AMT_CREDIT','AMT_ANNUITY','OCCUPATION_TYPE']].head(25)


CLIENTID  AMT_INCOME_TOTAL  AMT_CREDIT  AMT_ANNUITY OCCUPATION_TYPE
8      100011        112500.000   1019610.0      33826.5             NaN
11     100015         38419.155    148365.0      10678.5             NaN
23     100027         83250.000    239850.0      23850.0             NaN
28     100033        270000.000    790830.0      57676.5             NaN
30     100035        292500.000    665892.0      24592.5             NaN
38     100045         99000.000    247275.0      17338.5             NaN
43     100050        108000.000    746280.0      42970.5             NaN
44     100051        202500.000    661702.5      48280.5             NaN
46     100053        202500.000    305221.5      17649.0             NaN
52     100060         76500.000    454500.0      14661.0             NaN
54     100062         81000.000    675000.0      32472.0             NaN
56     100064         67500.000    298728.0      15381.0             NaN
62     100073        324000.000   1130760.0      40189.5             NaN
63     100075        112500.000     95940.0      10462.5             NaN
76     100088        112500.000    135000.0       6750.0             NaN
79     100094        108000.000    113760.0       5301.0             NaN
81     100096         81000.000    252000.0      14593.5             NaN
82     100097        157500.000    760225.5      30280.5             NaN
84     100099        360000.000    733315.5      41076.0             NaN
88     100103         72000.000    450000.0      16965.0             NaN
94     100112        315000.000    953460.0      64107.0             NaN
97     100115         90000.000    315000.0      14683.5             NaN
99     100118         90000.000    675000.0      26541.0             NaN
105    100124        225000.000    448272.0      28647.0             NaN
106    100125        126000.000    263686.5      26208.0             NaN

In [15]:
#Task 5.2.3.3
#It is not missing only NaN because maybe no one came with them to borrow loan
CC_CLIENTRATING1= CCLoan.loc[CCLoan['REGION_RATING_CLIENT'].isnull()]
CC_CLIENTRATING1 [['CLIENTID','AMT_INCOME_TOTAL','AMT_CREDIT','AMT_ANNUITY','REGION_RATING_CLIENT']].head(25)

Empty DataFrame
Columns: [CLIENTID, AMT_INCOME_TOTAL, AMT_CREDIT, AMT_ANNUITY, REGION_RATING_CLIENT]
Index: []

In [16]:

#Task 5.2.4
#Checking next 34-43 column of missing data 

CC_CLIENTCITYRATING = CCLoan['REGION_RATING_CLIENT_W_CITY'].isnull()
print ('CC_CLIENTCITYRATING Missing: '+str(CC_CLIENTCITYRATING.sum()))

CC_WEEKDAY_APPR = CCLoan['WEEKDAY_APPR_PROCESS_START'].isnull()
print ('CC_WEEKDAY_APPR Missing: '+str(CC_WEEKDAY_APPR.sum()))

CC_HOUR_APPR=CCLoan['HOUR_APPR_PROCESS_START'].isnull()
print ('Gender Missing: '+str(CC_HOUR_APPR.sum()))

CC_REGNOTLIVE=CCLoan['REG_REGION_NOT_LIVE_REGION'].isnull()
print ('CC_HOUR_APPR Missing: '+str(CC_REGNOTLIVE.sum()))

CC_REGIONNOTWORK=CCLoan['REG_REGION_NOT_WORK_REGION'].isnull()
print ('CC_REGIONNOTWORK Missing: '+str(CC_OWNREALITY.sum()))

CC_LIVEREGION =CCLoan['LIVE_REGION_NOT_WORK_REGION'].isnull()
print ('CC_LIVEREGION Missing: '+str(CC_OWNREALITY.sum()))

CC_CITYNOTLIVE=CCLoan['REG_CITY_NOT_LIVE_CITY'].isnull()
print ('CC_CITYNOTLIVE Missing: '+str(CC_INCOME.sum()))

CC_CITYNOTWORK=CCLoan['REG_CITY_NOT_WORK_CITY'].isnull()
print ('CC_CITYNOTWORK Missing: '+str(CC_CREDIT.sum()))

CC_LIVECITY=CCLoan['LIVE_CITY_NOT_WORK_CITY'].isnull()
print ('CC_LIVECITY Missing: '+str(CC_ANNUITY.sum()))

CC_GOODSPRICE=CCLoan['ORGANIZATION_TYPE'].isnull()
print ('CC_GOODSPRICE Missing: '+str(CC_GOODSPRICE.sum()))


CC_CLIENTCITYRATING Missing: 0
CC_WEEKDAY_APPR Missing: 0
Gender Missing: 0
CC_HOUR_APPR Missing: 0
CC_REGIONNOTWORK Missing: 0
CC_LIVEREGION Missing: 0
CC_CITYNOTLIVE Missing: 0
CC_CITYNOTWORK Missing: 0
CC_LIVECITY Missing: 0
CC_GOODSPRICE Missing: 0


In [17]:
#Task 5.2.5
#Checking COLUMN 


CC_EXT_SOURCE_1 = CCLoan['EXT_SOURCE_1'].isnull()
print ('EXT_SOURCE_1 Missing: '+str(CC_Late.sum()))

CC_EXT_SOURCE_2 = CCLoan['EXT_SOURCE_2'].isnull()
print ('EXT_SOURCE_2 Missing: '+str(CC_CONTRACT.sum()))

EXT_SOURCE_3=CCLoan['EXT_SOURCE_3'].isnull()
print ('EXT_SOURCE_3 Missing: '+str(CC_GENDER.sum()))

CC_APARTMENTS_AVG=CCLoan['APARTMENTS_AVG'].isnull()
print ('APARTMENTS_AVG Missing: '+str(CC_OWNCAR.sum()))

CC_BASEMENTAREA_AVG=CCLoan['BASEMENTAREA_AVG'].isnull()
print ('BASEMENTAREA_AVG Missing: '+str(CC_OWNREALITY.sum()))

CC_DAYS_LAST_PHONE_CHANGE=CCLoan['DAYS_LAST_PHONE_CHANGE'].isnull()
print ('DAYS_LAST_PHONE_CHANGE Missing: '+str(CC_DAYS_LAST_PHONE_CHANGE.sum()))

#CC_CHILD =CCLoan['LIVE_REGION_NOT_WORK_REGION'].isnull()
#print ('Own Reality Missing: '+str(CC_OWNREALITY.sum()))

CC_CREDITMONTH=CCLoan['AMT_REQ_CREDIT_BUREAU_MON'].isnull()
print ('Credit Month Missing: '+str(CC_CREDITMONTH.sum()))

CC_CREDITQRT=CCLoan['AMT_REQ_CREDIT_BUREAU_QRT'].isnull()
print ('Credit Quarter Missing: '+str(CC_CREDIT.sum()))

CC_CREDITYEAR=CCLoan['AMT_REQ_CREDIT_BUREAU_YEAR'].isnull()
print ('Credit Year Missing: '+str(CC_CREDITYEAR.sum()))





EXT_SOURCE_1 Missing: 0
EXT_SOURCE_2 Missing: 0
EXT_SOURCE_3 Missing: 0
APARTMENTS_AVG Missing: 0
BASEMENTAREA_AVG Missing: 0
DAYS_LAST_PHONE_CHANGE Missing: 1
Credit Month Missing: 41519
Credit Quarter Missing: 0
Credit Year Missing: 41519


In [21]:
CCLoan['FLAG_OWN_CAR'].count()

307511

In [25]:
LatePay= CCLoan.loc[CCLoan['LATE_PAYMENT']==1]
NoLate = CCLoan.loc[CCLoan['LATE_PAYMENT']==0]

In [66]:
#percent of people who are late vs nonlate on cc payment
print ('% of Late Payment : ' +str(len(LatePay['CLIENTID'])/len(CCLoan['CLIENTID'])*100))
print ('% in good standing : ' +str(len(NoLate['CLIENTID'])/len(CCLoan['CLIENTID'])*100))


#there is a 8% default on credit


% of Late Payment : 8.072881945686495
% in good standing : 91.92711805431351


In [59]:
NoLate

CLIENTID  LATE_PAYMENT NAME_CONTRACT_TYPE CODE_GENDER FLAG_OWN_CAR  \
1         100003             0         Cash loans           F            N   
2         100004             0    Revolving loans           M            Y   
3         100006             0         Cash loans           F            N   
4         100007             0         Cash loans           M            N   
5         100008             0         Cash loans           M            N   
...          ...           ...                ...         ...          ...   
307505    456249             0         Cash loans           F            N   
307506    456251             0         Cash loans           M            N   
307507    456252             0         Cash loans           F            N   
307508    456253             0         Cash loans           F            N   
307510    456255             0         Cash loans           F            N   

       FLAG_OWN_REALTY  CNT_CHILDREN  AMT_INCOME_TOTAL  AMT_CREDIT  \
1                    N             0          270000.0   1293502.5   
2                    Y             0           67500.0    135000.0   
3                    Y             0          135000.0    312682.5   
4                    Y             0          121500.0    513000.0   
5                    Y             0           99000.0    490495.5   
...                ...           ...               ...         ...   
307505               Y             0          112500.0    225000.0   
307506               N             0          157500.0    254700.0   
307507               Y             0           72000.0    269550.0   
307508               Y             0          153000.0    677664.0   
307510               N             0          157500.0    675000.0   

        AMT_ANNUITY  ...  FLAG_DOCUMENT_18 FLAG_DOCUMENT_19 FLAG_DOCUMENT_20  \
1           35698.5  ...                 0                0                0   
2            6750.0  ...                 0                0                0   
3           29686.5  ...                 0                0                0   
4           21865.5  ...                 0                0                0   
5           27517.5  ...                 0                0                0   
...             ...  ...               ...              ...              ...   
307505      22050.0  ...                 0                0                0   
307506      27558.0  ...                 0                0                0   
307507      12001.5  ...                 0                0                0   
307508      29979.0  ...                 0                0                0   
307510      49117.5  ...                 0                0                0   

       FLAG_DOCUMENT_21 AMT_REQ_CREDIT_BUREAU_HOUR AMT_REQ_CREDIT_BUREAU_DAY  \
1                     0                        0.0                       0.0   
2                     0                        0.0                       0.0   
3                     0                        NaN                       NaN   
4                     0                        0.0                       0.0   
5                     0                        0.0                       0.0   
...                 ...                        ...                       ...   
307505                0                        0.0                       0.0   
307506                0                        NaN                       NaN   
307507                0                        NaN                       NaN   
307508                0                        1.0                       0.0   
307510                0                        0.0                       0.0   

        AMT_REQ_CREDIT_BUREAU_WEEK  AMT_REQ_CREDIT_BUREAU_MON  \
1                              0.0                        0.0   
2                              0.0                        0.0   
3                              NaN                        NaN   
4                              0.0              

In [58]:
#Task 6.1
#Find what is percentage who is Late Payment that owns a car

owncar_LatePayment=(len(LatePay[LatePay['FLAG_OWN_CAR']=='Y']))/len(LatePay['FLAG_OWN_CAR'])*100
print ('% whom has defaulted: '+ str(owncar_LatePayment))


owncar_NoLate=(len(NoLate[NoLate['FLAG_OWN_CAR']=='Y']))/len(NoLate['FLAG_OWN_CAR'])*100
print('% that has not defaulted : '+ str(owncar_NoLate))


#there is a 4% difference where person who owns car that didnt default as to to whom has

Percentage whom has defaulted: 30.517623363544814
% that has not defaulted : 34.317582052170955


In [ ]:
Reality
No Reality

In [ ]:
#Task 6.2
#Find what is percentage who is Late Payment that owns reality

owncar_LatePayment=(len(LatePay[LatePay['FLAG_OWN_CAR']=='Y']))/len(LatePay['FLAG_OWN_CAR'])*100
print ('% whom has defaulted: '+ str(owncar_LatePayment))


owncar_NoLate=(len(NoLate[NoLate['FLAG_OWN_CAR']=='Y']))/len(NoLate['FLAG_OWN_CAR'])*100
print('% that has not defaulted : '+ str(owncar_NoLate))



explorte data



In [ ]:
#
#should do a graph of client ID vs the amount AMT_GOOD_PRICE / AMT_CREDIT

In [ ]:
#
#should do a graph of client ID vs the amount AMT_CREDIT / AMT_INCOME_TOTAL

In [ ]:
#Task x.6.5
#total people with cars and then information if they are late or not late on payment
car_owners = CCLoan[CCLoan['FLAG_OWN_CAR']== 'Y']
car_owner_T = car_owners[car_owners['LATE_PAYMENT']== 1]
car_owner_NT = car_owners[car_owners['LATE_PAYMENT']== 0]
print('Total Count:    '+ str(car_owners['LATE_PAYMENT'].count()))  #.sum() properway
print('LatePayment:    '+ str(car_owner_T['LATE_PAYMENT'].count()))
print('No LatePayment: '+ str(car_owner_NT['LATE_PAYMENT'].count()))

In [ ]:
#Task 6
#Find the % that is late vs not late
#8.1 % of total client is late
VS = (Late['CLIENTID'].count() / CCLoan['CLIENTID'].count())*100

print(VS)


In [ ]:
#Task 7
#get a list of clients who have a late installment
Late = CCLoan.loc[CCLoan['LATE_PAYMENT']. isin([1])]

In [ ]:
#Task 8
#count number of Target are late ---24825 is the output
Late.count()

In [ ]:
#Task 9
#try to find relationship where 
Late.head(25)


In [ ]:
#Task 10
Late[['CLIENTID','AMT_INCOME_TOTAL','AMT_CREDIT','AMT_ANNUITY','AMT_GOODS_PRICE']].head(25)

In [ ]:
#Task 11
Late.describe()

In [ ]:
#Task 21
#put non late payment into another DF so I can graph the latepayment vs income
NonLate = CCLoan.loc[CCLoan['LATE_PAYMENT']. isin([0])]

In [ ]:
CCLoan[['CLIENTID','LATE_PAYMENT','AMT_INCOME_TOTAL','AMT_CREDIT','AMT_ANNUITY','AMT_GOODS_PRICE','FLAG_OWN_CAR']].head(25)

#Task 22
#graph amt_credit vs amt_annuity

y = NonLate['AMT_CREDIT']
y2 = NonLate['AMT_ANNUITY']

x = NonLate ['CLIENT']
fig = plt.figure()
ax = plt.subplot()
ax.plot(x, y, label='Credit Amount')
ax.plot(x, y2, label='Loan Payment')
plt.title('Legend inside')
ax.legend()


In [ ]:
#GRAPH target vs points FLAG_OWN_CAR, FLAG_OWN_REALTY




In [ ]:
#Task 1000
#testsplit  dataset
from sklearn.model_selection import train_test_split

train_test_split(x,y,test_size = .2)